include/qpid/dispatch/ctools.h

```diff
@@ -217,4 +218,41 @@ do {                                    \
 #define MIN(a,b) (((a)<(b))?(a):(b))
 #define MAX(a,b) (((a)>(b))?(a):(b))
 
+//
+// Heap allocation with abort() on failure
+//
+
+static inline void *qd_malloc(size_t size)
+{
+    void *ptr = malloc(size);
+    if (!ptr)
+        abort();
```

Comment:
maybe some error message for the abort? on the other hand, that abort will probably never be called in practice, so it does not matter for UX...

Reply:
I was thinking about that, but decided that if a hard memory limit is hit it is unlikely that doing any I/O would succeed.

But at this point it probably cannot hurt to call perror before aborting...  at best it will indicate the failure was due to low memory.   I'll update the patch.